fmriprep/cli/run.py

```diff
@@ -286,12 +286,17 @@ def before_send(event, hints):
             exec_env = os.name
             # special variable set in the container
             if os.getenv('IS_DOCKER_8395080871'):
-                # based on https://stackoverflow.com/a/42674935/616300
-                with open('/proc/1/cgroup', 'rt') as ifh:
-                    if 'docker' in ifh.read():
-                        exec_env = 'docker'
-                    else:
-                        exec_env = 'singularity'
+                docker_version = os.getenv('DOCKER_VERSION_8395080871')
```

Comment:
Sometimes more is more in tearms of readability. I find the original easier to understand, but it's your call.

Reply:
I agree with @chrisfilo on this one. I find @oesteban's suggestion requires more thinking through, and it's such peripheral functionality that making people work to understand it does not seem worth a reduction in branching.

Reply:
Can we agree at least on this bit?:

```Python
                if 'docker' in Path('/proc/1/cgroup').read_text():
                    exec_env = 'docker'
```

Other than that, I'm fine either way.

Reply:
```
else:
    exec_env = 'singularity'
```
?

You just want to avoid using `with`?

Reply:
Sure, that doesn't add to cognitive overhead.

Reply:
@chrisfilo yes, I like to avoid the context, but more importantly, to leverage `pathlib.Path` that is much more readable IMHO (and already imported).

But yes, I meant to keep the else branch:

```Python
                if 'docker' in Path('/proc/1/cgroup').read_text():
                    exec_env = 'docker'
                else:
                    exec_env = 'singularity'
```

Although I find this more readable, but would be coming back to square 1 and I totally agree with you both. Whatever you think has less cognitive load :+1: 

```Python
                exec_env = 'singularity'
                if 'docker' in Path('/proc/1/cgroup').read_text():
                    exec_env = 'docker'
```